### ISSUE TITLE
microg installation gets broken because of google support libraries update

### ISSUE THREAD

**Issue by patrickdrd** [microg/android_packages_apps_GmsCore#1109]
every morning at this same time, my microg installation goes nuts,
I think this is happening because google is trying to update its support libraries,
and it could have to do with my patched signature,
I mean google fails but manages to do enough in order to break my microg installation,
only solution is to install it over (version code is reported as "-1"!) and this happens only with the new version 0.2.11,
it wasn't happening with 0.2.10

**Comment by Axolord** [comment]
Hi, can you elaborate a bit more? What Android version are you on and what do you mean by "google fails" and "microg installation goes nuts"? What exactly is happening?

**Comment by patrickdrd** [comment]
I'm on nougat and, for example google calendar sync fails and google drive sync as well etc,
I meant my microg installation doesn't get replaced by google's gms but enough damage is done to break my microg in a way that it needs to be reinstalled

**Comment by Axolord** [comment]
Are you using Lineage4microG or what Rom are you on? Seems pretty odd to me, that this would happen. Please provide a logcat, there are multiple apps on the fdroid that can read it or you could use adb from a PC.

My guess would be, that the problem is specific to you

**Comment by patrickdrd** [comment]
I'm on aospextended rom, this problem isn't specific to me, my guess is that it is specific to those of us that have signature checking disabled,
maybe you should read over here:

https://www.andnixsh.com/2020/02/disable-signature-check-using-core.html

and once again this started happening with the latest build 0.2.11, 
anyway, happened again today, some minutes ago so I guess I have to rollback to 0.2.10 till this is fixed

my logcat is available but I don't know how to strip sensitive information, so I don't know what to do with it

**Comment by patrickdrd** [comment]
there's what happens after the support libraries update:

```
E/Drivesync(24796): Caused by: com.google.android.gms.auth.GooglePlayServicesAvailabilityException: Google Play Services not available
W/GooglePlayServicesUtil(2796): Google Play services out of date.  Requires 11717000 but found -1
```

**Comment by mar-v-in** [comment]
First off, this patch sounds terrible to me. It's a huge security risk and it's very likely to cause problems - like the one you are seeing. -1 is not a valid package version code, so your package manager reached a broken state, which is a bug of the (patched) system and not related to microG.

Have you had 0.2.10 installed on /system and just upgraded to 0.2.11? In that case it may be that the upgrade attempt uninstalls the 0.2.11 without properly installing anything else. A workaround then would be to overwrite 0.2.10 with 0.2.11 on /system directly without installing it as an update.

There is no change to microG in 0.2.11 that could be remotely related to this.

**Comment by patrickdrd** [comment]
yes, I've had an old version (0.2.6) of microGServicesCore.apk installed on priv-app,
which I removed, 
and now microg shows installed as user app,
is this fine now? or do I need to convert it to system app?

**Comment by mar-v-in** [comment]
microG can work fine as a user app under certain conditions, but I don't know if this is going to cause problems with your setup.

**Comment by patrickdrd** [comment]
sure, testing will tell, the problem is that I can't think of a way to simulate this "google play support libraries update",
it seems to happen once daily , every morning, so I would have to wait a couple of days to see which of the user/system would work for me

**Comment by patrickdrd** [comment]
sure, testing will tell, the problem is that I can't think of a way to simulate this "google play support libraries update",
it seems to happen once daily , every morning, so I would have to wait a couple of days to see which of the user/system would work for me

edit: meanwhile, I'm not exactly sure that signature patching is related to this, it's just speculation..

anyway, I'm trying to think what could be happening,
one thought is that this daily update is uninstalling 0.2.11 because it is a user app and finds 0.2.6 as a system app?
microg doesn't get completely uninstalled

so my first test would be to keep microg as a user app only, let's see what happens next time
